app/src/main/java/fr/free/nrw/commons/review/ReviewActivity.java

```diff
@@ -125,45 +118,34 @@ public boolean runRandomizer() {
         progressBar.setVisibility(View.VISIBLE);
         reviewPager.setCurrentItem(0);
         compositeDisposable.add(reviewHelper.getRandomMedia()
-                .map(Media::getFilename)
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(this::updateImage));
         return true;
     }
 
     @SuppressLint("CheckResult")
-    private void updateImage(String fileName) {
+    private void updateImage(Media media) {
+        String fileName = media.getFilename();
         if (fileName.length() == 0) {
             ViewUtil.showShortSnackbar(drawerLayout, R.string.error_review);
             return;
         }
-        simpleDraweeView.setImageURI(Utils.makeThumbBaseUrl(fileName));
+
+        simpleDraweeView.setImageURI(media.getImageUrl());
+
         reviewController.onImageRefreshed(fileName); //file name is updated
-        compositeDisposable.add(reviewHelper.getFirstRevisionOfFile("File:" + fileName)
+        compositeDisposable.add(reviewHelper.getFirstRevisionOfFile(fileName)
```

Comment:
I know this is beyond the scope of this PR, but can you clear the composite disposable in the onStop/ onDestroy of the activity, ?